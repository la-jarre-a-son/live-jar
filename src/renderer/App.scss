$ljas-ui-asset-path: 'assets';
$font-path: '../assets/fonts';

@import "@flaticon/flaticon-uicons/css/regular/rounded.css";
@import "@flaticon/flaticon-uicons/css/brands/all.css";

@import '@la-jarre-a-son/ui/lib/theme/jar/tokens';
@import '@la-jarre-a-son/ui/lib/style/normalize';
@import '@la-jarre-a-son/ui/lib/theme/jar/global';
@import '@la-jarre-a-son/ui/lib/theme/jar/fonts';
@import '@la-jarre-a-son/ui/lib/theme/jar/dark';

html,
body {
  height: 100%;
  width: 100%;
  user-select: none;
}

#root {
  width: 100%;
  height: 100%;
}

