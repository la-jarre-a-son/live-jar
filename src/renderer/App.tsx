import { RouterProvider } from 'react-router-dom';
import { ThemeProvider } from '@la-jarre-a-son/ui';

import ApiTwitchProvider from 'renderer/contexts/ApiTwitch';

import router from './router';

import WindowStateProvider from './contexts/WindowState';
import SettingsProvider from './contexts/Settings';

import './App.scss';

export default function App() {
  return (
    <ThemeProvider theme="jar" variant="dark">
      <WindowStateProvider>
        <SettingsProvider>
          <ApiTwitchProvider>
            <RouterProvider router={router} />
          </ApiTwitchProvider>
        </SettingsProvider>
      </WindowStateProvider>
    </ThemeProvider>
  );
}
