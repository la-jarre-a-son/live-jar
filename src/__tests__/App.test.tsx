import '@testing-library/jest-dom';
import { render, act } from '@testing-library/react';
import App from '../renderer/App';

describe('App', () => {
  it('should render', async () => {
    const el = document.createElement('div');
    await act(async () => {
      await render(<App />, { container: el });
    });

    expect(el).toMatchSnapshot();
  });
});
