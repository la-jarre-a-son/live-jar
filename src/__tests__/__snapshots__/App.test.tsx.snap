// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div>
  <div
    class="base"
  >
    <div
      class="root root --elevation3 root --relative --top topbar topbar--isMac"
    >
      <nav
        class="root"
      >
        <ol>
          <li>
            <a
              aria-label="Live Jar"
              class="item"
              data-discover="true"
              href="#/"
              role="link"
              tabindex="0"
            >
              <img
                alt=""
                height="32"
                src="test-file-stub"
                width="32"
              />
              <span
                class="label"
              >
                Live Jar
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              aria-label="Windows"
              class="item active"
              data-discover="true"
              href="#/window"
              role="link"
              tabindex="0"
            >
              <test-file-stub
                classname="base base--inherit icon--windows"
              />
              <span
                class="label"
              >
                Windows
              </span>
            </a>
          </li>
        </ol>
      </nav>
      <div
        class="titlebar"
      />
      <div
        class="root --horizontal root --horizontal"
      >
        <a
          aria-label="Settings"
          aria-pressed="false"
          class="root --icon --md --filled --neutral"
          data-discover="true"
          href="#/settings"
          type="button"
        >
          <test-file-stub
            classname="base base--inherit icon--settings"
          />
        </a>
        <button
          aria-label="Always on Top"
          aria-pressed="false"
          class="root --icon --md --filled --neutral"
          type="button"
        >
          <test-file-stub
            classname="base base--inherit icon--pin"
          />
        </button>
        <button
          aria-label="Quit App"
          class="root --icon --md --filled --danger --hoverIntent"
          type="button"
        >
          <test-file-stub
            classname="base base--inherit icon--power"
          />
        </button>
      </div>
    </div>
    <div
      class="modalContainer"
    >
      <div
        class="content"
      >
        <div
          class="base"
        >
          <div
            class="root --xl --center container"
          >
            <div
              class="root --gap-lg col--md"
            >
              <a
                class="root --outlined --elevation1 root --interactive --outlined"
                data-discover="true"
                href="#/window/new"
              >
                <div
                  class="thumbnail"
                >
                  <div
                    class="thumbnailWrapper"
                  >
                    <img
                      alt="Stream preview"
                      class=""
                    />
                    <div
                      class="thumbnailPlaceholder"
                      role="presentation"
                    />
                    <div
                      class="root --xl --round"
                    >
                      <test-file-stub
                        classname="base base--inherit icon--plus"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="header"
                >
                  <div
                    class="headerContent"
                  >
                    Add window
                  </div>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;
